MSFT_EXOTransportRule: IncidentReportContent value casting

<!--
    Your feedback and support is greatly appreciated, thanks for contributing!

    ISSUE TITLE:
    Please prefix the issue title with the resource name, e.g.
    'ResourceName: Short description of my issue'

    ISSUE DESCRIPTION (this template):
    Please provide information regarding your issue under each header below.
    Write N/A under any headers that do not apply to your issue, or if the
    information is not available.

    NOTE! Sensitive information should be obfuscated.

    PLEASE KEEP THE HEADERS.

    You may remove this comment block, and the other comment blocks,
    but please keep the headers.
-->
#### Details of the scenario you tried and the problem that is occurring
Manage an EXO Transport Rule that sends an Incident report to an email address with some of the available message properties. Defining these properties as a single string (single and double quotes) 'Sender,Recipients', "Sender,Recipients" and both produce an error: 
``` Cannot process argument transformation on parameter 'IncidentReportContent'. Cannot convert value "System.Collections.ArrayList" to type "Microsoft.Exchange.MessagingPolicies.Rules.IncidentReportContent[]". ```

Also tried to provide the value as an array string "Sender", "Recipients", but produces a different error which is noted in issue [1198](https://github.com/microsoft/Microsoft365DSC/issues/1198). For full insight, providing an array string as opposed to a single string produces the error when compiling the MOF file.
``` Convert property 'IncidentReportContent' value from type 'STRING[]' to type 'STRING' failed ``` 

#### Verbose logs showing the problem
```
VERBOSE: [hostname]: LCM:  [ Start  Resource ]  [[EXOTransportRule]PPForward]
VERBOSE: [hostname]: LCM:  [ Start  Test     ]  [[EXOTransportRule]PPForward]
VERBOSE: [hostname]:                            [[EXOTransportRule]PPForward] Testing Transport Rule configuration for ExampleTransportRuleName
VERBOSE: [hostname]:                            [[EXOTransportRule]PPForward] Getting Transport Rule configuration for ExampleTransportRuleName
VERBOSE: [hostname]:                            [[EXOTransportRule]PPForward] Found Transport Rule ExampleTransportRuleName
VERBOSE: [hostname]:                            [[EXOTransportRule]PPForward] Current Values: ## Omitted ##
VERBOSE: [hostname]:                            [[EXOTransportRule]PPForward] Target Values: ## Omitted ##
VERBOSE: [hostname]:                            [[EXOTransportRule]PPForward] Test-TargetResource returned False
VERBOSE: [hostname]: LCM:  [ End    Test     ]  [[EXOTransportRule]PPForward]  in 0.6250 seconds.
VERBOSE: [hostname]: LCM:  [ Start  Set      ]  [[EXOTransportRule]PPForward]
VERBOSE: [hostname]:                            [[EXOTransportRule]PPForward] Setting Transport Rule configuration for ExampleTransportRuleName
VERBOSE: [hostname]:                            [[EXOTransportRule]PPForward] Getting Transport Rule configuration for ExampleTransportRuleName
VERBOSE: [hostname]:                            [[EXOTransportRule]PPForward] Found Transport Rule ExampleTransportRuleName
VERBOSE: [hostname]:                            [[EXOTransportRule]PPForward] Transport Rule 'ExampleTransportRuleName' already exists, but needs updating.
VERBOSE: [hostname]:                            [[EXOTransportRule]PPForward] Setting Transport Rule ExampleTransportRuleName with values:  ## Omitted ##
##[error]Failed to start the DSC configuration. Error: Cannot process argument transformation on parameter 'IncidentReportContent'. Cannot convert value "System.Collections.ArrayList" to type "Microsoft.Exchange.MessagingPolicies.Rules.IncidentReportContent[]". Error: "Cannot convert value "Sender,Recipients,Subject,Cc,Bcc,Severity,Override,RuleDetections,FalsePositive,DataClassifications,IdMatch,AttachOriginalMail" to type "Microsoft.Exchange.MessagingPolicies.Rules.IncidentReportContent" due to enumeration values that are not valid. Specify one of the following enumeration values and try again. The possible enumeration values are "Sender,Recipients,Subject,Cc,Bcc,Severity,Override,RuleDetections,FalsePositive,DataClassifications,IdMatch,AttachOriginalMail"."
```
#### Suggested solution to the issue
I'm not quite sure what would be an acceptable solution, but perhaps this can be resolved by splitting the input string? In this case the input value can be empty (''), single ('Sender') or multiple ('Sender,Recipients').
```powershell
$NewTransportRuleParams.IncidentReportContent = $NewTransportRuleParams.IncidentReportContent -split ','

$SetTransportRuleParams.IncidentReportContent = $SetTransportRuleParams.IncidentReportContent -split ','
```

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
```powershell
EXOTransportRule 'PPForward' {
    # Authentication details
    #Credential                  = $CredsGlobalAdmin
    TenantId                     = $TenantId
    ApplicationId                = $ApplicationId
    CertificateThumbprint        = $CertificateThumbprint

    # Settings
    Name                         = "ExampleTransportRuleName"
    Priority                     = 5
    Mode                         = "Enforce"
    Comments                     = "Internal reference URL here"

    # If the Message
    SenderAddressLocation        = "Header"
    SentToScope                  = "NotInOrganization"

    HeaderContainsMessageHeader  = "x-ms-mail-operation-type"
    HeaderContainsWords          = @("Forward")

    HeaderMatchesMessageHeader   = "x-ms-mail-application"
    HeaderMatchesPatterns        = @("Azure Logic Apps", "Microsoft Power Apps", "Microsoft Power Automate")

    StopRuleProcessing           = $False

    # Do the following
    DeleteMessage                = $True
    GenerateIncidentReport       = "mail@example.com"
    IncidentReportContent        = 'Sender,Recipients,Subject,Cc,Bcc,Severity,Override,RuleDetections,FalsePositive,DataClassifications,IdMatch,AttachOriginalMail'
}
```

#### The operating system the target node is running
<!--
    Please provide as much as possible about the target node, for example
    edition, version, build and language.
    On OS with WMF 5.1 the following command can help get this information.

    Get-ComputerInfo -Property @(
        'OsName',
        'OsOperatingSystemSKU',
        'OSArchitecture',
        'WindowsVersion',
        'WindowsBuildLabEx',
        'OsLanguage',
        'OsMuiLanguages')
-->
```
OsName               : Microsoft Windows Server 2019 Datacenter
OsOperatingSystemSKU : DatacenterServerEdition
OsArchitecture       : 64-bit
WindowsVersion       : 1809
WindowsBuildLabEx    : 17763.1.amd64fre.rs5_release.180914-1434
OsLanguage           : en-US
OsMuiLanguages       : {en-US} 
```
#### Version of the DSC module that was used ('dev' if using current dev branch)
1.22.720.1

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,9 @@
 
 # UNRELEASED
 
+* EXOTransportRule
+  * Fix issue setting IncidentReportContent
+  FIXES [#2196](https://github.com/microsoft/Microsoft365DSC/issues/2196)
 * AADConditionalAccessPolicy
   * DEPRECATED then IncludeDevices and ExcludeDevices parameters.
   * Fixed issue extracting a policy that had invalid users or groups (deleted from AAD).

--- a/Modules/Microsoft365DSC/DSCResources/MSFT_EXOTransportRule/MSFT_EXOTransportRule.psm1
+++ b/Modules/Microsoft365DSC/DSCResources/MSFT_EXOTransportRule/MSFT_EXOTransportRule.psm1
@@ -483,7 +483,7 @@ function Get-TargetResource
 
         [Parameter()]
         [System.String[]]
-        $IncidentReportContent,
+        $IncidentReportContent = @(),
 
         [Parameter()]
         [System.String[]]
@@ -1447,7 +1447,7 @@ function Set-TargetResource
 
         [Parameter()]
         [System.String[]]
-        $IncidentReportContent,
+        $IncidentReportContent = @(),
 
         [Parameter()]
         [System.String[]]
@@ -2230,7 +2230,7 @@ function Test-TargetResource
 
         [Parameter()]
         [System.String[]]
-        $IncidentReportContent,
+        $IncidentReportContent = @(),
 
         [Parameter()]
         [System.String[]]

--- a/Modules/Microsoft365DSC/DSCResources/MSFT_EXOTransportRule/MSFT_EXOTransportRule.schema.mof
+++ b/Modules/Microsoft365DSC/DSCResources/MSFT_EXOTransportRule/MSFT_EXOTransportRule.schema.mof
@@ -117,7 +117,7 @@ class MSFT_EXOTransportRule : OMI_BaseResource
     [Write, Description("The HeaderContainsWords parameter specifies a condition that looks for words in a header field.")] String HeaderContainsWords[];
     [Write, Description("The HeaderMatchesMessageHeader parameter specifies the name of header field in the message header when searching for the text patterns specified by the HeaderMatchesPatterns parameter.")] String HeaderMatchesMessageHeader;
     [Write, Description("The HeaderMatchesPatterns parameter specifies a condition that looks for text patterns in a header field by using regular expressions. ")] String HeaderMatchesPatterns[];
-    [Write, Description("The IncidentReportContent parameter specifies the message properties that are included in the incident report that's generated when a message violates a DLP policy. ")] String IncidentReportContent;
+    [Write, Description("The IncidentReportContent parameter specifies the message properties that are included in the incident report that's generated when a message violates a DLP policy. ")] String IncidentReportContent[];
     [Write, Description("The IncidentReportOriginalMail parameter specifies whether to include the original message with the incident report. ")] String IncidentReportOriginalMail;
     [Write, Description("The ManagerAddresses parameter specifies the users (managers) for the ExceptIfManagerForEvaluatedUser parameter.")] String ManagerAddresses[];
     [Write, Description("The ManagerForEvaluatedUser parameter specifies a condition that looks for users in the Manager attribute of senders or recipients."), ValueMap{"Recipient","Sender"}, Values{"Recipient","Sender"}] String ManagerForEvaluatedUser;

